### ISSUE TITLE
Using winpthreads with python leads to confliciting `pid_t`

### ISSUE THREAD

**Issue by tobiasdiez** [mingw-w64/mingw-w64#65]
On a 64-windows system, I get the the following error when I use winpthreads but also include Python:

```
C:\Users\Tobia\.conda\envs\sage-dev\Library\include\pthread_compat.h(74): error C2371: 'pid_t': redefinition; different basic types
C:\Users\Tobia\.conda\envs\sage-dev\Include\pyconfig.h(202): note: see declaration of 'pid_t'
```

The python config header has `typedef int pid_t;` independent of 32 or 64bit windows.

**Comment by Biswa96** [comment]
Are you using winpthreads with MSVC toolchain? I'm not sure if that's a supported use case.

**Comment by lhmouse** [comment]
Indeed on 64-bit Linux, `pid_t` is still a 32-bit value. However I am uncertain about what should we do... Changing `pid_t` to `int` for 64-bit targets would be an ABI break.

**Comment by tobiasdiez** [comment]
> Are you using winpthreads with MSVC toolchain? I'm not sure if that's a supported use case.

Indirectly yes: I'm using including a library (flint) from conda that makes use of winpthreads, but the main project is built using MSVC.

> Indeed on 64-bit Linux, `pid_t` is still a 32-bit value. However I am uncertain about what we should do... Changing `pid_t` to `int` for 64-bit targets would be an ABI break.

Would it be possible to check for an existing definition of `pid_t`, and only define it in `pthread_compat` in the absence of such?

**Comment by lhmouse** [comment]
> > Indeed on 64-bit Linux, `pid_t` is still a 32-bit value. However I am uncertain about what we should do... Changing `pid_t` to `int` for 64-bit targets would be an ABI break.
> 
> Would it be possible to check for an existing definition of `pid_t`, and only define it in `pthread_compat` in the absence of such?

I see your code defines its own `pid_t` in pyconfig.h, so it's likely that a 64-bit type breaks python, everywhere... the winpthreads `pid_t` shouldn't have been defined that way. (the native process ID is `DWORD` i.e. `unsigned long`, a 32-bit type.)

**Comment by tobiasdiez** [comment]
I agree. For completeness, here is the definition in Python: https://github.com/python/cpython/blob/19e93e2e269889ecb3c4c039091abff489f247c2/PC/pyconfig.h.in#L233
